test/functorch/test_ops.py

```diff
@@ -434,27 +425,66 @@ def test_jvp(self, device, dtype, op):
         inplace_variant = op.inplace_variant if op.supports_inplace_autograd else None
 
         for sample in samples:
+            if noncontiguous:
+                sample = sample.noncontiguous()
             args = (sample.input,) + sample.args
             kwargs = sample.kwargs
             if outplace_variant:
                 self.jvp_opinfo_test(outplace_variant, args, kwargs,
                                      sample.output_process_fn_grad,
                                      clone_inputs=False,
-                                     fixme_ref_jvp_local=fixme_ref_jvp_local)
+                                     fixme_ref_jvp_local=fixme_ref_jvp_local,
+                                     noncontiguous=noncontiguous)
             if is_valid_inplace_sample_input(sample, op, inplace_variant):
                 self.jvp_opinfo_test(inplace_variant, args, kwargs,
                                      sample.output_process_fn_grad,
                                      clone_inputs=True,
-                                     fixme_ref_jvp_local=fixme_ref_jvp_local)
+                                     fixme_ref_jvp_local=fixme_ref_jvp_local,
+                                     noncontiguous=noncontiguous)
+
+    @ops(op_db + additional_op_db, allowed_dtypes=(torch.float,))
+    @skipOps('TestOperators', 'test_jvp', jvp_fail)
+    @opsToleranceOverride('TestOperators', 'test_jvp', (
+        tol1('nn.functional.conv_transpose3d',
+             {torch.float32: tol(atol=1e-04, rtol=1.3e-06)}, device_type='cuda'),
+        tol1('nn.functional.binary_cross_entropy_with_logits',
+             {torch.float32: tol(atol=4e-04, rtol=4e-04)}),
+        tol1('nn.functional.batch_norm',
+             {torch.float32: tol(atol=4e-05, rtol=5e-05)}),
+    ))
+    def test_jvp(self, device, dtype, op):
+        self._jvp_helper(device, dtype, op, noncontiguous=False)
+
+    @ops(op_db + additional_op_db, allowed_dtypes=(torch.float,))
+    @skipOps('TestOperators', 'test_jvp_noncontiguous', jvp_fail.union({
+        # This is expected to fail as the operator
+        # expects last dim to have stride=1
+        xfail('view_as_complex'),
+        # BUG
+        # RuntimeError: !self.requires_grad() || self.is_contiguous()
+        # INTERNAL ASSERT FAILED
+        xfail('as_strided_scatter'),
+    }))
+    @opsToleranceOverride('TestOperators', 'test_jvp_noncontiguous', (
+        tol1('nn.functional.conv_transpose3d',
+             {torch.float32: tol(atol=1e-04, rtol=1.3e-06)}, device_type='cuda'),
+        tol1('nn.functional.binary_cross_entropy_with_logits',
+             {torch.float32: tol(atol=4e-04, rtol=4e-04)}),
+        tol1('nn.functional.batch_norm',
+             {torch.float32: tol(atol=4e-05, rtol=5e-05)}),
+    ))
+    def test_jvp_noncontiguous(self, device, dtype, op):
+        self._jvp_helper(device, dtype, op, noncontiguous=True)
```

Comment:
It might be better to combine test_jvp and test_jvp_noncontiguous. From experience, people are not going to remember to add xfails to the jvp_fail list; instead, they will add separate xfails to both tests separately. 

In general I'm fine with saying that "we don't have jvp support for the operator if it doesn't pass both tests".

The only case that is annoying is the view_as_complex case where only one of the two tests is legitimate, but it's only one special case.

Reply:
Done :)